### PR TITLE
Internal: AI builder screenshots pagination [ED-11536]

### PR DESCRIPTION
**Pull request by rotemee**

### CODE REVIEW

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js [hunk 1]
```diff
@@ -44,16 +42,33 @@ const useScreenshots = ( { onGeneration } ) => {
 						const updatedData = [ ...prev ];
 						const placeholderIndex = updatedData.lastIndexOf( PLACEHOLDER_VALUE );
 
-						updatedData.splice( placeholderIndex, 1 );
+						updatedData[ placeholderIndex ] = { isPlaceholder: true };
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js:45]: you have `PLACEHOLDER_VALUE` and `{ isPlaceholder: true }`
what's the difference? maybe change the name to one f them?

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+
+const useSlider = ( {
+	slidesCount = 0,
+	slidesPerPage = 3,
+	gapPercentage = 2,
+	translateXFilter = ( value ) => value,
+} = {} ) => {
+	const [ currentPage, setCurrentPage ] = useState( 1 );
+	const gapsNumber = slidesPerPage - 1;
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js:10]:
```suggestion
	const gapsCount = slidesPerPage - 1;
```

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	const translateX = translateXFilter( `-${ offsetXPercentage }%` );
+	const pagesCount = Math.ceil( slidesCount / slidesPerPage );
+	const sliderStyle = { overflow: 'hidden' };
+	const trackStyle = {
+		display: 'flex',
+		transition: 'all 0.4s ease',
+		gap: `${ gapPercentage }%`,
+		transform: `translateX(${ translateX })`,
+	};
+	const slideStyle = { flex: `0 0 ${ slideWidthPercentage }%` };
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js:22]: why the style needs to come from the hook?
can the hook have only the logic?

File: modules/ai/assets/js/editor/pages/form-layout/index.js [hunk 4]
```diff
@@ -86,6 +102,13 @@ const FormLayout = ( { onClose, onInsert, onGeneration, onSelect, DialogHeaderPr
 		lastRun.current();
 
 		setIsPromptEditable( false );
+		setCurrentPage( 1 );
+	};
+
+	const handleRegenerate = () => {
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/pages/form-layout/index.js:108]: if you have `handleRegenerate` maybe change the `handleSubmit` to `handleGenerate`?

File: modules/ai/assets/js/editor/pages/form-layout/index.js [hunk 5]
```diff
@@ -86,6 +102,13 @@ const FormLayout = ( { onClose, onInsert, onGeneration, onSelect, DialogHeaderPr
...
+		setCurrentPage( pagesCount + 1 );
+		regenerate( promptInputRef.current.value );
+		onRegenerate();
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/pages/form-layout/index.js:111]: hmm
I probably need an event in the onGenerate actually (right before sending the requests), so I'll be able to clear the previous containers on each new prompt

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js (hunk 2 above)
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js:10]: Done.

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js (hunk 3 above)
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/hooks/use-slider.js:22]: Done.

File: modules/ai/assets/js/editor/pages/form-layout/index.js (hunk 4 above)
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/index.js:108]: Done.

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js (hunk 1 above)
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js:45]: Done.

**Review by StyleShit — APPROVED** [review]

**Pull request merged**